.circleci/config.yml

```diff
@@ -28,7 +50,30 @@ jobs:
             - gradle-{{ checksum "build.gradle.kts" }}
             - gradle-
       - run:
-          command: ./gradlew check
+          command: |
+            cd src/test/kotlin
+            
+            # Find all Kotlin files that contain tests, write the filenames to filenames.txt and tell CircleCI to split them by filesize across the parallel executors
+            circleci tests glob "**/*.kt" \
+            | xargs grep -l -E '@Test|@ParameterizedTest' \
+            | cut -c 1- \
+            | circleci tests run --command=">filenames.txt xargs echo" \
+            --verbose \
+            --split-by=filesize
```

Comment:
Would have preferred to split by timings for a more even split - unfortunately not possible because JUnit doesn't always report times against class names where @Nested or @DisplayName has been used.

Reply:
Will also investigate splitting the tests by package name and having a parallel executor for each. This would require handrolling an alternative to `circleci tests glob` but should give us a more intuitive solution and make it easier to spot naughty packages.